sklearn/metrics/_pairwise_distances_reduction/_base.pyx

```diff
@@ -185,11 +184,10 @@ cdef class PairwiseDistancesReduction64:
         by parallelizing computation on the inner loop on chunks of Y
         and reduce them.
 
-        This strategy dispatches chunks of Y uniformly on threads.
-        Each thread processes all the chunks of X in turn. This strategy is
-        a sequence of embarrassingly parallel subtasks (the inner loop on Y
-        chunks) with intermediate datastructures synchronisation at each
-        iteration of the sequential outer loop on X chunks.
+        This strategy is a sequence of embarrassingly parallel subtasks:
+        it dispatches chunks of Y uniformly on threads for each chunk of X at a time.
```

Comment:
This is a bit more verbose, but imo makes it a bit clearer.

```suggestion
        chunks of X are iterated over sequentially, and for each chunk of X it dispatches threads on uniform chunks of Y.
```

Reply:
With the rephrasing proposed above (i.e https://github.com/scikit-learn/scikit-learn/pull/23978/files#r929593552).
```suggestion
        chunks of X are iterated over sequentially, and for each chunk
        of X, tasks are dispatched statically on threads.
        Each task processes one and only one chunk of Y,
        computing and reducing distances matrices between
        vectors of the chunk of X and vectors of the Y.
```